### PR TITLE
Use the same implementation of getAxisInfo for rank and contour plot

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

To follow up #656 

## What does this implement/fix? Explain your changes.

Use the same implementation of getAxisInfo for rank and contour plot

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@not522 Could you review this PR?

File: optuna_dashboard/ts/components/GraphRank.tsx [hunk 1]
```diff
@@ -153,35 +146,67 @@ const getRankPlotInfo = (
...
-      null
-    const yValue =
-      trial.params.find((p) => p.name === yAxis.name)?.param_external_value ||
-      null
-    if (trial.values === undefined || xValue === null || yValue === null) {
-      return
+  filteredTrials.forEach((trial, i) => {
+    if (xAxis.values[i] && yAxis.values[i] && trial.values) {
+      const xValue = xAxis.values[i] as string | number
+      const yValue = yAxis.values[i] as string | number
```
Comment (HideakiImamura) [inline optuna_dashboard/ts/components/GraphRank.tsx:156]: How about avoiding the type assertion as follows?

```suggestion
    const xValue = xAxis.values[i]
    const yValue = yAxis.values[i]
    if (xValue && yValue && trial.values) {
```

File: optuna_dashboard/ts/components/GraphRank.tsx [hunk 2]
```diff
@@ -153,35 +146,67 @@ const getRankPlotInfo = (
...
+    const yIndices: number[] = Array.from(Array(yValues.length).keys()).sort(
+      (a, b) =>
+        yValues[a]
+          .toString()
+          .toLowerCase()
+          .localeCompare(yValues[b].toString().toLowerCase())
+    )
+    xValues = yIndices.map((i) => xValues[i])
+    yValues = yIndices.map((i) => yValues[i])
+  }
```
Comment (HideakiImamura) [inline optuna_dashboard/ts/components/GraphRank.tsx:182]: Why are these two logics inconsistent?

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have two comments. PTAL.

File: optuna_dashboard/ts/components/GraphRank.tsx (hunk 2 above)
Comment (keisuke-umezawa) [reply optuna_dashboard/ts/components/GraphRank.tsx]: Sorry, it was my mistake and fixed it.

File: optuna_dashboard/ts/components/GraphRank.tsx (hunk 1 above)
Comment (keisuke-umezawa) [reply optuna_dashboard/ts/components/GraphRank.tsx]: Fixed it.

**Comment by HideakiImamura** [comment]
The test fails with optuna==2.10.0. Could you check it?

**Comment by keisuke-umezawa** [comment]
@HideakiImamura 
Sorry for being late. I fixed errors. Could you review it again?

**Review by not522 — APPROVED** [review]
Thank you for your PR! LGTM!

**Review by HideakiImamura — APPROVED** [review]
LGTM. I confirmed that the graph is appropriately generated.

**Pull request merged**